libraries/NDP/src/NDP.cpp

```diff
@@ -239,24 +213,39 @@ int NDPClass::sendData(uint8_t *data, unsigned int len)
   return s;
 }
 
-
+int NDPClass::extractStart(void) {
+  uint32_t sample_size;
+  int s = syntiant_ndp120_tiny_get_recording_metadata(ndp,
+      &sample_size, SYNTIANT_NDP120_GET_FROM_MCU);
+  return s;
+}
 
 int NDPClass::extractData(uint8_t *data, unsigned int *len) {
   int s;
-  unsigned int l = audio_sample_chunk_size;
-  s = syntiant_ndp120_tiny_send_audio_extract(ndp, SYNTIANT_NDP120_EXTRACT_FROM_UNREAD);
+  unsigned int l = 768;
+  // extract audio + annotations
+  s = syntiant_ndp120_tiny_extract_data(ndp, data, &l, 1);
+  // in this simple case, don't return annotations to user
   if (s) {
     *len = 0;
     return s;
   }
-  // extract audio + annotations
-  s = syntiant_ndp120_tiny_extract_data(ndp, data, &l);
-  // in this simple case, don't return annotations to user
-  l -= sizeof(syntiant_ndp120_tiny_dsp_audio_sample_annotation_t);
```

Comment:
Haven't checked if it's the case, but we need to make sure we are not returning the annotations to the user